src/engine/execute.js

```diff
@@ -558,6 +565,16 @@ const execute = function (sequencer, thread) {
             }
         }
     }
+
+    if (runtime.profiler !== null) {
+        if (blockCached._profiler !== runtime.profiler) {
+            prepareBlockProfiling(runtime.profiler, blockCached);
+        }
+        const end = Math.min(i + 1, length);
```

Comment:
Can you add a comment about what this calculation is? It's difficult to keep track of units and what `i` and `length` correspond to.

Reply:
Added.

Reply:
```js
// Determine the index that is after the last executed block. `i` is
// currently the block that was just executed. `i + 1` will be the block
// after that. `length` with the min call makes sure we don't try to
// reference an operation outside of the set of operations.
```